packages/turbo-utils/src/getTurboConfigs.ts

```diff
@@ -109,7 +120,89 @@ export function getTurboConfigs(cwd?: string, opts?: Options): TurboConfigs {
   }
 
   if (cacheEnabled && cwd) {
-    configsCache[cwd] = configs;
+    turboConfigsCache[cwd] = configs;
+  }
+
+  return configs;
+}
+
+export function getWorkspaceConfigs(
```

Comment:
The package detection in JS should be ported from the Rust version. I've written this a bunch of times now and the Rust one is the most-correct implementation we've ever had.

This right here was a workaround to make sure we actually tracked workspaces correctly.